// @dapi-co/dapi-node/lib/types/models/Currency.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.XAU = exports.USD = exports.TRY = exports.TND = exports.CHF = exports.SEK = exports.LKR = exports.ZAR = exports.SGD = exports.SAR = exports.OMR = exports.QAR = exports.PHP = exports.GBP = exports.PKR = exports.NOK = exports.NZD = exports.MAD = exports.KWD = exports.CZK = exports.JOD = exports.JPY = exports.INR = exports.HKD = exports.EUR = exports.EGP = exports.DKK = exports.CAD = exports.BHD = exports.AED = exports.AUD = void 0;
exports.AUD = { code: 'AUD', name: 'AUSTRALIAN DOLLAR' };
exports.AED = {
    code: 'AED',
    name: 'UNITED ARAB EMIRATES DIRHAM',
};
exports.BHD = { code: 'BHD', name: 'BAHRAINI DINAR' };
exports.CAD = { code: 'CAD', name: 'CANADIAN DOLLAR' };
exports.DKK = { code: 'DKK', name: 'DANISH KRONE' };
exports.EGP = { code: 'EGP', name: 'EGYPTIAN POUND' };
exports.EUR = { code: 'EUR', name: 'EURO' };
exports.HKD = { code: 'HKD', name: 'HONG KONG DOLLAR' };
exports.INR = { code: 'INR', name: 'INDIAN RUPEE' };
exports.JPY = { code: 'JPY', name: 'JAPANESE YEN' };
exports.JOD = { code: 'JOD', name: 'JORDANIAN DINAR' };
exports.CZK = { code: 'CZK', name: 'KORONA' };
exports.KWD = { code: 'KWD', name: 'KUWAITI DINAR' };
exports.MAD = { code: 'MAD', name: 'MOROCCAN DIRHAM' };
exports.NZD = { code: 'NZD', name: 'NEW ZEALAND DOLLAR' };
exports.NOK = { code: 'NOK', name: 'NORWEGIAN KRONE' };
exports.PKR = { code: 'PKR', name: 'PAKISTAN RUPEE' };
exports.GBP = { code: 'GBP', name: 'POUND STERLING' };
exports.PHP = { code: 'PHP', name: 'PHILIPPINE PESO' };
exports.QAR = { code: 'QAR', name: 'QATARI RIYAL' };
exports.OMR = { code: 'OMR', name: 'RIAL OMANI' };
exports.SAR = { code: 'SAR', name: 'SAUDI RIYAL' };
exports.SGD = { code: 'SGD', name: 'SINGAPORE DOLLAR' };
exports.ZAR = { code: 'ZAR', name: 'SOUTH AFRICAN RAND' };
exports.LKR = { code: 'LKR', name: 'SRI LANKA RUPEE' };
exports.SEK = { code: 'SEK', name: 'SWEDISH KRONA' };
exports.CHF = { code: 'CHF', name: 'SWISS FRANC' };
exports.TND = { code: 'TND', name: 'TUNISIAN DINARS' };
exports.TRY = { code: 'TRY', name: 'TURKISH LIRA' };
exports.USD = { code: 'USD', name: 'US DOLLAR' };
exports.XAU = { code: 'XAU', name: 'GOLD' };
